patches/server/0969-New-specific-event-watching-command.patch

```diff
@@ -0,0 +1,196 @@
+         } catch (final Throwable e) {
+diff --git a/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java b/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1a063e467daa91178adf85f2351faf64692d5055
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java
+@@ -0,0 +1,131 @@
++package io.papermc.paper.command.subcommands;
++
++import com.google.gson.JsonArray;
++import com.google.gson.JsonObject;
++import com.google.gson.internal.Streams;
++import com.google.gson.stream.JsonWriter;
++import io.papermc.paper.command.PaperSubcommand;
++import io.papermc.paper.util.misc.EventWrapper;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.Event;
++import org.bukkit.event.Listener;
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.checkerframework.framework.qual.DefaultQualifier;
++
++import java.io.IOException;
++import java.io.PrintStream;
++import java.io.StringWriter;
++import java.nio.charset.StandardCharsets;
++import java.nio.file.Files;
++import java.nio.file.Path;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.ArrayList;
++import java.util.List;
++import java.util.Locale;
++import java.util.Map;
++import java.util.Collections;
++import java.util.concurrent.ConcurrentHashMap;
++
++import static net.kyori.adventure.text.Component.text;
++import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
++import static net.kyori.adventure.text.format.NamedTextColor.RED;
++
++@DefaultQualifier(NonNull.class)
++public class EventWatcherCommand implements PaperSubcommand {
++
++    /**
++     * The key is the equivalent of {@link Class#getName()} on the class.
++     * The left side of the pair value is what file we're going to write to, and
++     */
++    private static final Map<String, List<Wrapper>> watching = new ConcurrentHashMap<>(); //only concurrent in the rare case that multiple people are debugging, should be unnecessary
++    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH.mm.ss");
++    private static final Path PARENT = Path.of("event-debug");
++
++
++    public static void tryLog(Event event, @Nullable Listener listener) { //listener should only be null on the initial event call
++        List<Wrapper> wrappers = watching.get(event.getClass().getName());
++        if (wrappers != null)
++            wrappers.add(new Wrapper(event, listener));
++    }
++
++    @Override
++    public boolean execute(CommandSender sender, String subCommand, String[] args) {
++        if (args.length < 1) {
++            sender.sendMessage(text("Invalid Usage! /paper watchevent [event name]", RED));
```

Comment:
Generally `<event name>` is the syntax for required arguments while `[event name]` means its optional.